R/fit_SE.R

```diff
@@ -15,6 +15,8 @@ fit_SE <- function(se,
 
   # Assertions:
   checkmate::assert_class(se, "SummarizedExperiment")
+  checkmate::assert_string(data_type)
+  checkmate::assert_choice(data_type, c("single-agent", "combination"))
```

Comment:
Or just simpler? :) 
```
checkmate::assert_string(data_type, pattern = "single-agent|combination")
```

Reply:
Other question - is here "combination" correct value?
1. There is a separate function `fit_SE.combination` (for `data_type = "matrix"`)?
2. Param `data_type` is used in `gDRcore::data_model()`.  So if fit_SE is meant for `"single-agent"` type data - `data_type %in% c("single-agent", "cotreatment", "co-dilution")`, right?

Reply:
First of all I like the idea of using regular expressions over hardcoded values but I feel like mine version is safer and easier to change. For instance `checkmate::assert_string(data_type, pattern = "single-agent|combination")` accepts `data_type = "acombination"` since the pattern checks out but is not a meaningful value of this argument. Writing proper regular expression will be IMO time consuming.
Second of all, I agree with the combination. I will use `gDRcore:::get_data_type_to_data_model_mapping()` function (or `gDRutils::get_experiment_groups()`). Thanks!